### DISCUSSION TITLE
ask smapi update-skill-manifest error: DENIED_FEATURE_ACCESS for "AppLink V2" feature

### DISCUSSION THREAD

**Discussion by markapeters** [aws/aws-cli#8879] [Q&A]
I'm trying to create an Alexa for apps custom skill to allow voice control of my Android app. The app's deep links are verified and working and the skill itself is partially implemented in the Alexa Developer Console (so AWS account created along with IAM setup for root user). The ASK CLI tool is installed on my Windows 11 desktop and apparently working.

I've written a skills.json file, which by everything I know is correct, yet when trying to update it using:

`ask smapi update-skill-manifest -s SKILLID -g development --manifest "file:skill.json" --debug`

I get a 400 error with the following response body:

`{"message":"Skill manifest is not valid.","violations":[{"code":"DENIED_FEATURE_ACCESS","message":"User is not allowed access to the \"AppLink V2\" feature and, therefore, cannot use string instance with value \"\"APP_LINKS_V2\"\" at \"$.manifest.apis.custom.interfaces[0].type\". Further information: \"alexaforapps-help@amazon.com\".","validationDetails":{"feature":{"contact":"alexaforapps-help@amazon.com","name":"AppLink V2"},"originalInstance":{"dataType":"string","propertyPath":"$.manifest.apis.custom.interfaces[0].type","type":"BODY","value":"\"APP_LINKS_V2\""}}}]}`

I thought perhaps my IAM configuration is missing a permission, but I haven't found a permission change/addition that looks relevant.

Does anybody know what might be causing this error? For what it's worth, I found that the tutorials available for setting up AWS and IAM very hard to follow since the steps shown and documented in them are nothing like the current UI available at AWS. It is entirely possible that I missed something important.

**Comment by markapeters** [comment]
It turns out that the reason the error happens is that Amazon has dropped support for Alexa for Apps, so it is no longer possible to develop a skill for that. The docs claim that it is still available using command line tools, but I suspect the docs are just out of date.

**Comment by RyanFitzSimmonsAK** [comment]
Hey @markapeters, thanks for reaching out.  The ASK CLI and the AWS CLI are actually totally different.  It looks like they do have a public Github repository, so if you want the documentation to be addressed, you could cut them an issue (https://github.com/alexa/ask-cli/issues).
